src/main/java/org/veupathdb/eda/binaryfiles/printer/BinaryFilePrinter.java

```diff
@@ -89,8 +90,39 @@ private static void printAncestorFile(Path binaryFile, JSONObject metajson) {
     }
   }    
   
-  private static void printVarFile(Path binaryFile, BinaryConverter<?> converter) {
+  private static void printIdsMapFile(Path binaryFile, JSONObject metajson) {
+
+    int numAncestors = metajson.getInt(BinaryFilesManager.META_KEY_NUM_ANCESTORS);
+
+    IdsMapConverter converter = new IdsMapConverter(numAncestors);
+    
+    try (DualBufferBinaryRecordReader reader = new DualBufferBinaryRecordReader(binaryFile,
+                                                                                        converter.numBytes(), RECORDS_PER_BUFFER)){
+
+        while (true) {
+          Optional<byte[]> rowBytes = reader.next();
+          if (rowBytes.isEmpty())
+            break;
+          IdsMap idsMapRow = rowBytes.map(converter::fromBytes)
+            .orElseThrow(() -> new RuntimeException("Unexpected end of ancestors file"));
```

Comment:
You can just call `.get()` instead of `orElseThrow()` since you've already confirmed it's not empty at this point.